vendor/engines/split_accounts/app/services/split_accounts/order_detail_list_transformer.rb

```diff
@@ -0,0 +1,25 @@
+module SplitAccounts
+
+  class OrderDetailListTransformer
+
+    attr_reader :order_details
+
+    def initialize(order_details = [])
+      @order_details = order_details
+    end
+
+    def perform
+      order_details.map do |order_detail|
```

Comment:
I'll put this in a separate PR once this is in since it will cause conflicts: I was doing some performance testing. Testing against 1000 OrderDetails all being on a SplitAccounts, using the following using `each_with_object` instead of the `map`/`flatten` had a 30% speed increase (~7.2 seconds down to ~4.7). For reference, testing against 1000 OrderDetails on a normal account, there is negligible performance impact of using this Transformer.

```ruby
      def perform
        order_details.each_with_object([]) do |order_detail, results|
          # We will need to refactor the general_reports_controller_spec in
          # order to remove the `try` methods below.
          if order_detail.account.try(:splits).try(:present?)
            results.concat SplitAccounts::OrderDetailSplitter.new(order_detail).build_split_order_details
          else
            results << order_detail
          end
        end
      end
```